File: fstream-core/src/main/java/io/fstream/core/service/StateService.java
/*
 * Copyright (c) 2015 fStream. All Rights Reserved.
 *
 * Project and contact information: https://bitbucket.org/fstream/fstream
 *
 * Unauthorized copying of this file, via any medium is strictly prohibited.
 * Proprietary and confidential.
 */

package io.fstream.core.service;

import io.fstream.core.model.state.State;
import io.fstream.core.model.state.StateListener;

public interface StateService {

  public abstract void initialize();

  public abstract void destroy();

  public abstract State getState();

  public abstract void setState(State state);

  public abstract void addListener(StateListener listener);

}

File: fstream-ui/app/scripts/services/state-service.js
angular
   .module('fstream')
   .factory('stateService', ['$http', stateService]);

function stateService($http) {
	return {
		getState: function() {
			return $http.get('/state').then(function(response) {
				return response.data;
			});              
		}
	} 
}
